extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/ColorSplasher.pushbutton/script.py

```diff
@@ -164,133 +142,136 @@ def Execute(self, uiapp):
-                new_id_legend = legends[0].Duplicate(ViewDuplicateOption.Duplicate)
+                new_id_legend = legends[0].Duplicate(DB.ViewDuplicateOption.Duplicate)
                 newLegend = new_doc.GetElement(new_id_legend)
-                i=1
+                i = 1
                 while True:
                     try:
-                        newLegend.Name = "Color Legend " + str(i)
+                        if rvt_ver < 2025:
+                            newLegend.Name = unicode("Color Legend " + str(i))
+                        else:
+                            newLegend.Name = "Color Legend " + str(i)
                         break
                     except:
-                        i+=1
+                        i += 1
                         if i == 1000:
                             break
                 trans.Commit()
-                old_all_ele = FilteredElementCollector(new_doc, legends[0].Id).ToElements()
-                ele_id_type = ElementId(0)
+                old_all_ele = DB.FilteredElementCollector(new_doc, legends[0].Id).ToElements()
+                ele_id_type = DB.ElementId(0)
                 for ele in old_all_ele:
-                    if ele.Id != newLegend.Id and ele.Category != None:
-                        if isinstance(ele, TextNote):
+                    if ele.Id != newLegend.Id and ele.Category is not None:
+                        if isinstance(ele, DB.TextNote):
                             ele_id_type = ele.GetTypeId()
                             break
                 get_elementid_value = get_elementid_value_func()
                 if get_elementid_value(ele_id_type) == 0:
-                     #Get any text in model
-                    all_text_notes = FilteredElementCollector(new_doc).OfClass(TextNoteType).ToElements()
+                    # Get any text in model
+                    all_text_notes = DB.FilteredElementCollector(new_doc).OfClass(DB.TextNoteType).ToElements()
                     for ele in all_text_notes:
                         ele_id_type = ele.Id
-                        break                  
+                        break
                 sel_cat = wndw._categories.SelectedItem['Value']
                 sel_par = wndw._listBox1.SelectedItem['Value']
                 list_max_X = []
                 list_y = []
-                #FilledRegionType
+                # FilledRegionType
                 filled_type=[]
-                all_types = FilteredElementCollector(new_doc).OfClass(FilledRegionType).ToElements()
+                all_types = DB.FilteredElementCollector(new_doc).OfClass(DB.FilledRegionType).ToElements()
                 for ty in all_types:
                     pattern = new_doc.GetElement(ty.ForegroundPatternId)
-                    if pattern != None:
+                    if pattern is not None:
                         if pattern.GetFillPattern().IsSolidFill:
                             if ty.ForegroundPatternColor.IsValid:
```

Comment:
these can be made into a single `if` condition with `and` (and maybe turned into a `is_valid_solid_fill` function or something like that)